staging/src/k8s.io/pod-security-admission/admission/admission.go

```diff
@@ -319,7 +319,7 @@ func (a *Admission) ValidatePod(ctx context.Context, attrs Attributes) *admissio
 		klog.ErrorS(err, "failed to fetch pod namespace", "namespace", attrs.GetNamespace())
 		return internalErrorResponse(fmt.Sprintf("failed to lookup namespace %s", attrs.GetNamespace()))
 	}
-	nsPolicy, _ := a.PolicyToEvaluate(namespace.Labels)
+	nsPolicy, nsPolicyErr := a.PolicyToEvaluate(namespace.Labels)
```

Comment:
Audit & warn default to privileged when they fail to parse, so if the namespace is non-enforcing it would shortcircuit and the parse error annotation would never be added.

What about doing something like this instead?
```go
if nsPolicyErr != nil {
  defer func() {
    if response == sharedAllowedResponse() {
      response = allowedResponse() // Don't touch the shared response.
    }
    if response.AuditAnnotations == nil {
      response.AuditAnnotations = map[string]string{}
    }
    response.AuditAnnotations["error"] = fmt.Sprintf("Failed to parse policy: %v", nsPolicyErr)
  }
}
```

Reply:
I don't really want to get in the business of comparing to the `sharedAllowedResponse` pointer value... that seems fragile. I expect the nsPolicyErr case to be vanishingly rare (requires someone to have set malformed labels on a namespace prior to the admission plugin being enabled), so I just limited the short-circuit to the non-error case.